### PR TITLE
Update reqs to 1.0

### PR DESCRIPTION
**Pull request by austingmhuang**
This new device will no longer support Qiskit <0.46. Therefore testing for those versions is no longer necessary and has been deleted from the CI workflow.

### CODE REVIEW

**Comment by austingmhuang** [comment]
[sc-63543]

**Review by lillian542 — COMMENTED** [review]
Looks good! A few small loose ends to wrap up:

1. lets also rename `tests_qiskit_1.yml` to `tests.yml`, since that is the standard name people will be looking for to find the test CI file
2. there are also 0.46 and 1.0 tests run twice a week for the test matrix (the IBMQ tests) - we should similarly delete the 0.46 version and rename the 1.0 version to `ibmq_tests.yml`. Long term those tests might not make sense as we move to local testing with FakeHardware, but for now let's just do that.
3. Lets add a comment on the PR for the feature branch targeting master, to remind us to update the Plugin Test Matrix when the branch is eventually merged - removing the IBMQ tests will need to be reflected [here](https://github.com/PennyLaneAI/plugin-test-matrix) (but not until the tests are removed on `master`)

**Comment by lillian542** [comment]
Wrt codecov, it's just upset that initializing the `BasicAer` device isn't being tested. 

We can wait to merge this until we've merged a PR removing the `BasicAer` device, and then re-trigger CI and it should pass. Alternatively, we can have someone with authorization override CodeCov and merge.

**Comment by austingmhuang** [comment]
> We can wait to merge this until we've merged a PR removing the `BasicAer` device, and then re-trigger CI and it should pass. Alternatively, we can have someone with authorization override CodeCov and merge.

I think we can wait and merge this after removing the `BasicAer` device. I'll open that PR ASAP.

**Comment by austingmhuang** [comment]
> Looks good! A few small loose ends to wrap up:
> 
> 1. lets also rename `tests_qiskit_1.yml` to `tests.yml`, since that is the standard name people will be looking for to find the test CI file

Ok yeah, I didn't know that :sweat_smile: 

> 2. there are also 0.46 and 1.0 tests run twice a week for the test matrix (the IBMQ tests) - we should similarly delete the 0.46 version and rename the 1.0 version to `ibmq_tests.yml`. Long term those tests might not make sense as we move to local testing with FakeHardware, but for now let's just do that.

Ok, makes sense 👍 


> 3. Lets add a comment on the PR for the feature branch targeting master, to remind us to update the Plugin Test Matrix when the branch is eventually merged - removing the IBMQ tests will need to be reflected [here](https://github.com/PennyLaneAI/plugin-test-matrix) (but not until the tests are removed on `master`)

Done.

File: .github/workflows/ibmq_tests.yml
Comment (obliviateandsurrender) [inline .github/workflows/ibmq_tests.yml:4]: why `01:00`? 🤔

File: .github/workflows/tests.yml
Comment (obliviateandsurrender) [inline .github/workflows/tests.yml:82]: Is there any performance degradation for `basicsim` in comparison to `basicaer`?

**Review by obliviateandsurrender — COMMENTED** [review]
About the tests being removed, will we be still testing for `0.46` or removing them fully? 🤔

File: .github/workflows/ibmq_tests.yml
Comment (austingmhuang) [reply .github/workflows/ibmq_tests.yml:4]: Oh actually I don't know. I deleted the old tests and renamed this one and I guess they had a different schedule? If it matters I can always change it back 😅

File: .github/workflows/tests.yml
Comment (austingmhuang) [reply .github/workflows/tests.yml:82]: `Basicaer` is deprecated in >1.0, and `basicsim` is the 1.0 replacement for it. Shouldn't be any performance degradation as far as I can tell.

**Comment by austingmhuang** [comment]
> About the tests being removed, will we be still testing for `0.46` or removing them fully? 🤔

We will be removing them fully. `0.46` and previous versions will no longer be supported.
TL;DR: In October Qiskit will no longer support 0.46 and previous versions accessing the real hardware, so it doesn't make sense for us to support it on the July release just to deprecate it in October.

File: tests/test_qiskit_device.py
Comment (austingmhuang) [reply tests/test_qiskit_device.py:179]: I can change the test so that it checks the message text still

**Review by lillian542 — APPROVED** [review]
Looks good to me, thanks @austingmhuang !

**Review by obliviateandsurrender — APPROVED** [review]
:shipit:

**Pull request merged**
